docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,814 @@
+If you are receiving Interactions over the gateway, you will **also need to respond via HTTP**. Responses to Interactions **are not sent as commands over the gateway**.
+
+To respond to a gateway Interaction, make a `POST` request like this. `interaction_id` is the unique id of that individual Interaction from the receieved payload. `interaction_token` is the unique token for that interaction from the received payload. **This endpoint is only valid for Interactions received over the gateway. Otherwise, respond to the `POST` request.**
+
+```py
+url = "https://discord.com/api/v8/interactions/<interaction_id>/<interaction_token>/callback"
+
+json = {
+    "type": 1,
+    "data": {
+        "content": "Congrats on sending your command!"
+    }
+}
+r = requests.post(url, json=json)
+```
+
+> info
+> Interaction `tokens` are valid for **15 minutes** and can be used to send followup messages.
+
+## Followup Messages
+
+Sometimes, your bot will want to send followup messages to a user after responding to an interaction. Or, you may want to edit your original response. You can use the following endpoints to do so:
+
+- `PATCH /webhooks/<application_id>/<interaction_token>/messages/@original` to edit your initial response to an Interaction
+- `DELETE /webhooks/<application_id>/<interaction_token>/messages/@original` to delete your initial response to an Interaction
+- `POST /webhooks/<application_id>/<interaction_token>` to send a new followup message
+- `PATCH /webhooks/<application_id>/<interaction_token>/messages/<message_id>` to edit a message sent with that `token`
+
+> danger
+> Interactions webhooks share the same rate limit properties as normal webhooks
+
+Interaction tokens are valid for **15 minutes**, meaning you can respond and update to a Slash Command within that amount of time.
+
+## Security and Authorization
+
+> info
+> Check out our [Community Resources](#DOCS_TOPICS_COMMUNITY_RESOURCES/interactions) for libraries to help you with security in your language of choice
+
+The internet is a scary place, especially for people hosting open, unauthenticated endpoints. If you are receiving Interactions via outgoing webhook, there are some security steps you **must** take before your app is eligible to receive requests.
+
+Every Interaction is sent with the following headers:
+
+- `X-Signature-Ed25519` as a signature
+- `X-Signature-Timestamp` as a timestamp
+
+Using your favorite security library, you **must validate the request each time to you receive an interaction**. If the signature fails validate, respond with a `401` error code. Here's a couple code examples:
+
+```js
+// This example uses the nobe-ed25519 node package
+// MY_CLIENT_PUBLIC_KEY for your app can be found on the General Information page for your app in the Dev Portal
+
+ const signature = req.get('X-Signature-Ed25519');
+ const timestamp = req.get('X-Signature-Timestamp');
+ const isValidRequest = await verifyKey(req.rawBody, signature, timestamp, 'MY_CLIENT_PUBLIC_KEY');
```

Comment:
let's recommend tweetnacl since it's more popular and has been audited.

```
const nacl = require('tweetnacl');

const PUBLIC_KEY = 'APPLICATION_PUBLIC_KEY';

const signature = req.get('X-Signature-Ed25519');
const timestamp = req.get('X-Signature-Timestamp');
const body = req.rawBody; // rawBody is expected to be a string, not raw bytes

const isVerified = nacl.sign.detached.verify(
  Buffer.from(timestamp + body),
  Buffer.from(signature, 'hex'),
  Buffer.from(PUBLIC_KEY, 'hex')
);
```